src/main/java/org/opentripplanner/routing/graphfinder/PlaceFinderTraverseVisitor.java

```diff
@@ -176,8 +179,22 @@ private void handleStop(RegularStop stop, double distance) {
       !seenStops.contains(stop.getId()) &&
       (filterByModes == null || stopHasPatternsWithMode(stop, filterByModes))
     ) {
-      placesFound.add(new PlaceAtDistance(stop, distance));
-      seenStops.add(stop.getId());
+      if (
+        includeStopsAndStations &&
+        stop.getParentStation() != null &&
+        !seenStops.contains(stop.getParentStation().getId()) &&
+        !seenStops.contains(stop.getId())
+      ) {
+        seenStops.add(stop.getParentStation().getId());
+        placesFound.add(new PlaceAtDistance(stop.getParentStation(), distance));
+      } else {
+        // Do not accumulate the same station many times
+        if (
+          stop.getParentStation() != null && seenStops.contains(stop.getParentStation().getId())
+        ) return;
```

Comment:
Try to always wrap things inside an if in a {} block and preferable also split that block to a new line as it's easy to miss something happening on the same line where there is an if statement.